ui/apps/platform/src/Containers/Collections/CollectionForm.tsx

```diff
@@ -23,17 +23,53 @@ import {
 import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
 import useSelectToggle from 'hooks/patternfly/useSelectToggle';
 import { collectionsBasePath } from 'routePaths';
-import { ResolvedCollectionResponse } from 'services/CollectionsService';
 import { CollectionPageAction } from './collections.utils';
 import RuleSelector from './RuleSelector';
 import CollectionAttacher from './CollectionAttacher';
 import CollectionResults from './CollectionResults';
+import { Collection, ScopedResourceSelector, SelectorEntityType } from './types';
+
+type FormStateReducerAction =
+    | { type: 'setName'; name: string }
+    | { type: 'setDescription'; description: string }
+    | { type: 'setRules'; entity: SelectorEntityType; selector: ScopedResourceSelector }
+    | { type: 'attachCollection'; collectionId: string }
+    | { type: 'detachCollection'; collectionId: string };
+
+function formStateReducer(state: Collection, payload: FormStateReducerAction): Collection {
```

Comment:
This felt like a good place for a unit test, but I figured I would keep this un-exported since all of these use cases would be well tested by e2e tests once we get to that point.